skyportal/tests/frontend/test_shifts.py

```diff
@@ -259,16 +263,27 @@ def test_shift(
     driver.get("/shifts")
 
     # check for API shift
-    driver.wait_for_xpath(
-        shift_on_calendar,
-        timeout=30,
-    )
-
-    driver.click_xpath(shift_on_calendar)
+    shift_element = driver.wait_for_xpath(shift_on_calendar, timeout=30)
+    for i in range(30):  # stroll a few pixels to the left
+        try:
+            ActionChains(driver).move_to_element(shift_element).move_by_offset(
+                -i, 0
+            ).click().perform()
+            title = driver.wait_for_xpath('//*[@id="current_shift_title"]')
+            if title.text == form_name:
+                break
+        except UnexpectedAlertPresentException:
+            pass  # avoid popup asking to make new shift
+
+    assert title.text == form_name  # we found the right shift
 
     delete_button_xpath = '//*[@id="delete_button"]'
-    driver.wait_for_xpath(delete_button_xpath)
-    driver.click_xpath(delete_button_xpath)
+    driver.wait_for_xpath(delete_button_xpath).click()
+    try:  # try again
+        driver.wait_for_xpath(delete_button_xpath, timeout=1).click()
+    except TimeoutException:
```

Comment:
Why do we need to wait twice? Can't we set the timeout to be longer?

Reply:
I've just noticed that sometimes the click doesn't register. So if the button is still around the second time (with a timeout=1) will get it.